lib/portage/util/env_update.py

```diff
@@ -237,63 +238,16 @@ def _env_update(makelinks, target_root, prev_mtimes, contents, env, writemsg_lev
             if os.path.basename(y) != "libexec":
                 potential_lib_dirs.add(y[len(eroot) :])
 
-    # Update prelink.conf if we are prelink-enabled
-    if prelink_capable:
-        prelink_d = os.path.join(eroot, "etc", "prelink.conf.d")
-        ensure_dirs(prelink_d)
-        newprelink = atomic_ofstream(os.path.join(prelink_d, "portage.conf"))
-        newprelink.write(
-            "# prelink.conf autogenerated by env-update; make all changes to\n"
+    if not prelink_capable:
+        etc_path = Path(eroot, "etc")
+        prelink_files = (
+            etc_path.joinpath("prelink.conf.d", "portage.conf"),
+            etc_path.joinpath("prelink.cache"),
+            etc_path.joinpath("prelink.conf"),
         )
-        newprelink.write("# contents of /etc/env.d directory\n")
-
-        for x in sorted(potential_lib_dirs) + ["bin", "sbin"]:
-            newprelink.write("-l /%s\n" % (x,))
-        prelink_paths = set()
-        prelink_paths |= set(specials.get("LDPATH", []))
-        prelink_paths |= set(specials.get("PATH", []))
-        prelink_paths |= set(specials.get("PRELINK_PATH", []))
-        prelink_path_mask = specials.get("PRELINK_PATH_MASK", [])
-        for x in prelink_paths:
-            if not x:
-                continue
-            if x[-1:] != "/":
-                x += "/"
-            plmasked = 0
-            for y in prelink_path_mask:
-                if not y:
-                    continue
-                if y[-1] != "/":
-                    y += "/"
-                if y == x[0 : len(y)]:
-                    plmasked = 1
-                    break
-            if not plmasked:
-                newprelink.write("-h %s\n" % (x,))
-        for x in prelink_path_mask:
-            newprelink.write("-b %s\n" % (x,))
-        newprelink.close()
-
-        # Migration code path.  If /etc/prelink.conf was generated by us, then
-        # point it to the new stuff until the prelink package re-installs.
-        prelink_conf = os.path.join(eroot, "etc", "prelink.conf")
-        try:
-            with open(
-                _unicode_encode(
-                    prelink_conf, encoding=_encodings["fs"], errors="strict"
-                ),
-                "rb",
-            ) as f:
-                if (
-                    f.readline()
-                    == b"# prelink.conf autogenerated by env-update; make all changes to\n"
-                ):
-                    f = atomic_ofstream(prelink_conf)
-                    f.write("-c /etc/prelink.conf.d/*.conf\n")
-                    f.close()
-        except IOError as e:
-            if e.errno != errno.ENOENT:
-                raise
+        removable_files = (file for file in prelink_files if file.is_file())
+        for removable_file in removable_files:
+            removable_file.unlink()
```

Comment:
Maybe also try to remove `/etc/prelink.conf.d` directory (it might remain as empty directory), and silently ignore failure.